src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -906,10 +906,28 @@ private bool IsPossibleAttributeDeclaration()
 
         private SyntaxList<AttributeListSyntax> ParseAttributeDeclarations(bool inExpressionContext)
         {
-            var attributes = _pool.Allocate<AttributeListSyntax>();
             var saveTerm = _termState;
             _termState |= TerminatorState.IsAttributeDeclarationTerminator;
 
+            // An attribute can never appear *inside* an attribute declaration in the language.  However, during parsing
+            // we can end up in a state where we're trying to exactly that, through a path of
+            // Attribute->Argument->Expression->Attribute (since attributes can not be on lambda expressions).
```

Comment:
>(since attributes can not be on lambda expressions) [](http://example.com/codeflow?start=58&length=51)

Attributes can be applied to lambda expressions.

Reply:
> Attributes can be applied to lambda expressions.

Not legally.  That's my point here.  Attributes can't even actually be used within an attribute argument itself.